src/test/java/hudson/plugins/mercurial/PipelineTest.java

```diff
@@ -172,6 +185,69 @@
         assertFalse(iterator.hasNext());
     }
 
+    @Test public void modernHook() throws Exception {
+        String instName = "caching";
+        MercurialInstallation installation = new MercurialInstallation(instName, "", "hg", false, true, false, null,
+                Collections.<ToolProperty<?>>emptyList());
+        LogTaskListener listener = new LogTaskListener(Logger.getLogger(getClass().getName()), Level.INFO);
+        final HgExe hg = new HgExe(installation, null, r.jenkins.createLauncher(
+                listener), r.jenkins, listener, new EnvVars());
+        String version = hg.version();
+        assumeThat("Need mercurial 3.0 to have in-process hooks, have " + version, new VersionNumber(version).isNewerThan(new VersionNumber("3.0")),is(true));
+
+        sampleRepo.init();
+        ScriptApproval sa = ScriptApproval.get();
+        sampleRepo.write("Jenkinsfile", "node {checkout scm; echo readFile('file')}");
+        sampleRepo.write("file", "initial content");
+        sampleRepo.hg("commit", "--addremove", "--message=flow");
+        WorkflowMultiBranchProject mp = r.jenkins.createProject(WorkflowMultiBranchProject.class, "p");
+        r.jenkins.getDescriptorByType(MercurialInstallation.DescriptorImpl.class).setInstallations(installation);
+        installation.forNode(r.jenkins, new LogTaskListener(Logger.getAnonymousLogger(), Level.FINE));
```

Comment:
What are you trying to do here? Did you mean to use `StreamTaskListener.fromStdout()`?